src/__tests__/unit/audit.mixin.unit.ts

```diff
@@ -0,0 +1,574 @@
+import {Constructor} from '@loopback/core';
+import {expect, sinon} from '@loopback/testlab';
+import {
+  Action,
+  AuditLog,
+  AuditRepositoryMixin,
+  IAuditMixinOptions,
+} from '../..';
+import {
+  MockClass,
+  mockClassMethodCall,
+  resetMethodCalls,
+} from './fixtures/mockClass';
+import {MockModel} from './fixtures/mockModel';
+import {EntityCrudRepository} from '@loopback/repository';
+import {mockData, mockDataArray, resetMockData} from './fixtures/mockData';
+import {consoleMessage} from '../acceptance/audit.mixin.acceptance';
+
+let auditData: AuditLog;
+let auditDataArray: AuditLog[];
+let auditCreateCalled = false;
+let auditCreateAllCalled = false;
+
+class MockAuditRepo {
+  create(audit: AuditLog) {
+    auditData = audit;
+    auditCreateCalled = true;
+    return Promise.resolve('Audit Created');
+  }
+  createAll(auditArray: AuditLog[]) {
+    auditDataArray = auditArray;
+    auditCreateAllCalled = true;
+    return Promise.resolve('Audit Created');
+  }
+}
+
+class MockAuditRepoError {
+  create(audit: AuditLog) {
+    auditCreateCalled = true;
+    return Promise.reject('Error');
+  }
+  createAll(auditArray: AuditLog[]) {
+    auditCreateAllCalled = true;
+    return Promise.reject('Error');
+  }
+}
+
+const mockOpts: IAuditMixinOptions = {
+  actionKey: 'Test_Logs',
+};
+const mockUser = {
+  id: 'testCurrentUserId',
+  name: 'testCurrentUserName',
+};
+
+describe('Audit Mixin', () => {
+  const returnedMixedClass = AuditRepositoryMixin<
```

Comment:
Do all this in a method that is described at the end of the test. and call that method in the before hook